django_afip/signals.py

```diff
@@ -13,5 +14,31 @@ def update_certificate_expiration(sender, instance: models.TaxPayer, **kwargs):
 
 @receiver(post_save, sender=models.ReceiptPDF)
 def generate_receipt_pdf(sender, instance: models.ReceiptPDF, **kwargs):
-    if not instance.pdf_file and instance.receipt.is_validated:
-        instance.save_pdf(save_model=True)
+    if not instance.pdf_file:
+        if "CAEA" in instance.receipt.point_of_sales.issuance_type:
+            instance.save_pdf(save_model=True)
+        if (
+            "CAE" in instance.receipt.point_of_sales.issuance_type
+            and instance.receipt.is_validated
+        ):
+            instance.save_pdf(save_model=True)
+
+
+@receiver(pre_save, sender=models.Receipt)
+def save_caea_data(sender, instance: models.TaxPayer, **kwargs):
+    if "CAEA" in instance.point_of_sales.issuance_type:
+        if instance.caea == "" or instance.caea == None:
+            caea = models.Caea.objects.all().filter(active=True)
+            if caea.count() != 1:
+                raise exceptions.CaeaCountError
```

Comment:
I thought this in the way that you as user should implement a logic into your app to active/deactive CAEA to ensure this does not happen. Eg: a method that run as daemon that when the expiracy date of the CAEA arrive deactivate it.

Reply:
That requires the extra background daemon checking that.

Just filer by the correct period and order. See also: https://github.com/WhyNotHugo/django-afip/pull/164#discussion_r992465160

Reply:
In: [b284605](https://github.com/WhyNotHugo/django-afip/pull/164/commits/b2846057ec3c554f4d6003e876766a9f6ac1e3d1) changed the way that the caea is filtered.

Now in the way is implemented the only way that raise this error is if someone manually generate a caea with duplicated (period,order,active=True). For now I think that a custom manager for caea is too much, because you only filter in this place for a caea.